src/Symfony/Component/Console/Color.php

```diff
@@ -148,14 +148,35 @@ private function convertHexColorToAnsi(int $color): string
         $b = $color & 255;
 
         // see https://github.com/termstandard/colors/ for more information about true color support
-        if ('truecolor' !== getenv('COLORTERM')) {
-            return (string) $this->degradeHexColorToAnsi($r, $g, $b);
+        $termSupport = self::getTermColorSupport();
+
+        if ('Ansi4' === $termSupport) {
+            return (string) $this->degradeHexColorToAnsi4($r, $g, $b);
         }
 
+        // Ansi24 true color
         return sprintf('8;2;%d;%d;%d', $r, $g, $b);
     }
 
-    private function degradeHexColorToAnsi(int $r, int $g, int $b): int
+    public static function getTermColorSupport(): string
```

Comment:
Please define the public method before the private ones

Reply:
Maybe the right place for this static method can be on Terminal class instead ? Maybee it's a better place for Terminal capacity. (She need to be static for the tests. But the both degrade... methods should be statix too)

Reply:
The public static `getTermColorSupport` would indeed better fit in the Terminal class.

`degrade` methods don't need to become static. they are fine as is.